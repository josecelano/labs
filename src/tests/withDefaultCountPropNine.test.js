import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withDefaultCountPropNine from '../withDefaultCountPropNine';

const Counter = ({ count }) => (
  <div>Count: {count}</div>
)
const CounterWithDefaultNine = withDefaultCountPropNine(Counter);

it('should set the prop count to `9` by default', () => {
  const tree = toJson(shallow(<CounterWithDefaultNine />));
  expect(tree).toMatchSnapshot();
});

it('should accept a prop `count` which should overwrite the default', () => {
  const tree = toJson(shallow(<CounterWithDefaultNine count={12} />));
  expect(tree).toMatchSnapshot();
});
