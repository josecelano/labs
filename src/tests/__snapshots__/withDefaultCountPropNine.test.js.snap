exports[`test should accept a prop \`count\` which should overwrite the default 1`] = `
<Component
  count={12} />
`;

exports[`test should set the prop count to \`9\` by default 1`] = `
<Component
  count={9} />
`;
